From 8188c4a151ec066ac3d41cd40c9fa9f902754e28 Mon Sep 17 00:00:00 2001
From: Roman Li <Roman.Li@amd.com>
Date: Tue, 25 Jul 2017 14:50:44 -0400
Subject: [PATCH 2746/2831] drm/amd/display: Fix eDP panel light-up

 Power-down panel on disale only if sink exists
 otherwise it fails edid read during init sequence

Change-Id: Iefb7f16f7d465ad81ae83940237bc6a896429b29
Signed-off-by: Roman Li <Roman.Li@amd.com>
Reviewed-by: Andrey Grodzovsky <Andrey.Grodzovsky@amd.com>
Acked-by: Harry Wentland <Harry.Wentland@amd.com>
---
 drivers/gpu/drm/amd/display/dc/core/dc_link_hwss.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/amd/display/dc/core/dc_link_hwss.c b/drivers/gpu/drm/amd/display/dc/core/dc_link_hwss.c
index aac2420..5bdcd50 100644
--- a/drivers/gpu/drm/amd/display/dc/core/dc_link_hwss.c
+++ b/drivers/gpu/drm/amd/display/dc/core/dc_link_hwss.c
@@ -114,7 +114,9 @@ void dp_disable_link_phy(struct dc_link *link, enum signal_type signal)
 
 	if (signal == SIGNAL_TYPE_EDP) {
 		link->link_enc->funcs->backlight_control(link->link_enc, false);
-		link->link_enc->funcs->power_control(link->link_enc, false);
+		if (link->dpcd_sink_count)
+			link->link_enc->funcs->power_control(link->link_enc,
+									false);
 	}
 
 	link->link_enc->funcs->disable_output(link->link_enc, signal);
-- 
2.7.4

